File: csharp/DVSClient/DVSClient/common/ClientReference.cs
using System.Reflection;

namespace DVSClient.Common
{
    public class ClientReference
    {
        private const string PRODUCT = "SDK";
        private const string INTERFACE = "CSharp";

        // Private constructor to prevent instantiation
        private ClientReference() { }

        public static string GetReference(string uuid, bool allowsDotInReferenceId = true)
        {
            var assembly = Assembly.GetExecutingAssembly();
            var version = assembly?.GetName().Version?.ToString() ?? "0.0.0.0";

            // TODO once API has been updated.
            if (!allowsDotInReferenceId)
            {
                version = version.Replace('.', '-');
            }

            return $"product:{PRODUCT}/version:{version}/interface:{INTERFACE}/transaction:{uuid}";
        }
    }
}

File: java/src/main/java/com/experian/dvs/client/common/ClientReference.java
package com.experian.dvs.client.common;

public class ClientReference {

    private static final String PRODUCT = "SDK";
    private static final String INTERFACE = "Java";

    private ClientReference() {
        // Prevent instantiation
    }

    public static String getReference(final String uuid, boolean allowsDotInReferenceId) {
        String version = ClientReference.class.getPackage().getImplementationVersion();

        // TODO once API has been updated.
        if (version != null && !allowsDotInReferenceId)
        {
            version = version.replace('.', '-');
        }

        return String.format("product:%s/version:%s/interface:%s/transaction:%s", PRODUCT, version, INTERFACE, uuid);
    }
}
